x/mongo/driver/operation.go

```diff
@@ -217,6 +217,9 @@ type Operation struct {
 	// read preference will not be added to the command on wire versions < 13.
 	IsOutputAggregate bool
 
+	// MaxTime specifies the maximum amount of time to allow the operation to run on the server.
+	MaxTime *time.Duration
```

Comment:
Should we specify that this is specifically designed to work [using nanoseconds](https://pkg.go.dev/time#:~:text=A%20Duration%20represents%20the%20elapsed%20time%20between%20two%20instants%20as%20an%20int64%20nanosecond%20count)? This is kind of tricky and has taken me a while to wrap my head around -- but even though you can stick seconds, etc, in this field I think the results would be inaccurate (non-nanoseconds) for `(Operation).caclculateMaxTimeMS` because within that method we are assuming that the rtt90 and the `op.MaxTime` are in nanoseconds, given conversions such as `time/time.Milisecond`. Perhaps I'm misunderstanding something?

Reply:
[here](https://go.dev/play/p/D5trTdzdTql) is an example to illustrate the question.

Reply:
Sounds right to me, thank you for confirming!